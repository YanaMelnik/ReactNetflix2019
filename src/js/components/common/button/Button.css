.button {
    display: inline-block;
    padding: 5px 40px;
    font-family: Arial, serif;
    font-size: 12px;
    border-radius: 2px;
    color: white;
    font-weight: 600;
    margin: 10px 10px 30px;
    background-color: #f34c70;
    border: none;
    text-transform: uppercase;
}