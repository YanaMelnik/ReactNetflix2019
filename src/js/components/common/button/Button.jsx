import React, { Component } from 'react';
import PropTypes from 'prop-types';
import 'Button.css';

class Button extends Component {
  render() {
    return (
      <button
        className='button'
        onClick={this.props.action}>
        {this.props.name}
      </button>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default Button;
