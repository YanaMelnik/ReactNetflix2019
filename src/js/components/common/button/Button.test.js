import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    it('should render component correctly in case when all props defined', () => {
        const component = shallow(<Button name='test' action={() => {}}/>);

        expect(component).toMatchSnapshot();
    });

    it('should show btn name correctly', () => {
        const name = 'testName';
        const component = shallow(<Button name={name} action={() => {}}/>);

        expect(component.matchesElement(<button>{name}</button>)).toBeTruthy();
    });

    it('should call action func when user click on btn', () => {
        const mockAction = jest.fn();
        const component = shallow(<Button name='name' action={mockAction}/>);
        component.simulate('click');

        expect(mockAction.mock.calls.length).toBe(1);
    });
});
