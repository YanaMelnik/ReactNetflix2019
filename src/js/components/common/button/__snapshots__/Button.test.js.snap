// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component correctly in case when all props defined 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  test
</button>
`;
